### PR TITLE
tls: fix wrong pre_master_key length

### PR DESCRIPTION
**Pull request by azzbcc**

### CODE REVIEW

File: src/packet/packet_tls.h [hunk 1]
```diff
@@ -215,16 +215,11 @@ struct PreMasterSecret
...
-{
-    guint8 pre_master_secret[128];
-};
-
 //! ClientKeyExchange type in Handshake records
 struct ClientKeyExchange
 {
     uint16 length;
-    struct EncryptedPreMasterSecret exchange_keys;
+    guint8 pre_master_secret[];
```
Comment (Kaian) [inline src/packet/packet_tls.h:222]: I prefer to keepn these structures names and existance, even if they are not used.
They take the names from RFC4346, that defines:

```
      struct {
          public-key-encrypted PreMasterSecret pre_master_secret;
      } EncryptedPreMasterSecret;
```

Won't be enough to remove the 128 from EncryptedPreMasterSecret?

```
struct EncryptedPreMasterSecret
{
    guint8 pre_master_secret[];
};
```

**Review by Kaian — CHANGES_REQUESTED** [review]

File: src/packet/packet_tls.h (hunk 1 above)
Comment (azzbcc) [reply src/packet/packet_tls.h:222]: Tried but got flexible array member in a struct with no named members.

Comment (Kaian) [reply src/packet/packet_tls.h:222]: It's ok then, the other option is to use a pointer and allocate required memory, but that is a bit more complicated and can be done later.

File: src/packet/packet_tls.c [hunk 2]
```diff
@@ -638,12 +638,11 @@ static gboolean
...
     // Decrypt PreMasterKey
-    struct ClientKeyExchange clientkeyex;
-    memcpy(&clientkeyex, g_bytes_get_data(data, NULL), sizeof(struct ClientKeyExchange));
+    struct ClientKeyExchange *clientkeyex = (struct ClientKeyExchange *) g_bytes_get_data(data, NULL);
 
     gnutls_datum_t exkeys, pms;
-    exkeys.size = UINT16_INT(clientkeyex.length);
-    exkeys.data = (unsigned char *) &clientkeyex.exchange_keys;
+    exkeys.size = UINT16_INT(clientkeyex->length);
+    exkeys.data = (unsigned char *) &clientkeyex->pre_master_secret;
```
Comment (Kaian) [inline src/packet/packet_tls.c:645]: Feel free to change any std types for glib types everywhere you find them. In this case guchar pointer is perfectly fine.

**Review by Kaian — APPROVED** [review]

**Pull request merged**